### PR TITLE
add remote config support for sampling rules

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
https://docs.google.com/document/d/12lXVniuaYenanxnpt5gJJ76a1X7gFOfmCYVQVXlHNco/edit?usp=sharing

forcing system test to run with changes in .github/workflows/system-tests.yml, will delete the change after PR passes review

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/config.js [hunk 1]
```diff
@@ -870,6 +866,25 @@ class Config {
     }
   }
 
+  _setSamplingRule (obj, name, value) {
```
Comment (rochdev) [inline packages/dd-trace/src/config.js:869]: Why is this needed? It looks like cases where it should be a string VS an array are isolated.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/config.js (hunk 1 above)
Comment (ida613) [reply packages/dd-trace/src/config.js:869]: when it is null/undefined, we cannot do sample_rate -> sampleRate mapping on it

File: .github/workflows/system-tests.yml [hunk 2]
```diff
@@ -83,7 +83,7 @@ jobs:
       - name: Build
         run: ./build.sh -i runner
       - name: Run
-        run: ./run.sh PARAMETRIC
+        run: ./run.sh PARAMETRIC -F tests/parametric/test_dynamic_configuration.py::TestDynamicConfigSamplingRules
```
Comment (simon-id) [inline .github/workflows/system-tests.yml:86]: reminder to remove before merging :+1:

File: packages/dd-trace/src/appsec/remote_config/capabilities.js [hunk 3]
```diff
@@ -15,5 +15,6 @@ module.exports = {
   APM_TRACING_LOGS_INJECTION: 1n << 13n,
   APM_TRACING_HTTP_HEADER_TAGS: 1n << 14n,
   APM_TRACING_CUSTOM_TAGS: 1n << 15n,
-  APM_TRACING_ENABLED: 1n << 19n
+  APM_TRACING_ENABLED: 1n << 19n,
+  APM_TRACING_SAMPLE_RULES: 1n << 29n
```
Comment (simon-id) [inline packages/dd-trace/src/appsec/remote_config/capabilities.js:19]: you're adding a new capability but you're never using it ?

File: packages/dd-trace/src/config.js [hunk 4]
```diff
@@ -972,6 +968,25 @@ class Config {
...
+    if (value === null || value === undefined) {
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:972]: FYI this is equivalent to:
```suggestion
    if (value == null) {
```
up to you to decide which one you prefere

File: packages/dd-trace/src/config.js [hunk 5]
```diff
@@ -972,6 +968,25 @@ class Config {
...
+      return this._setValue(obj, name, null)
+    }
+
+    if (typeof value === 'string') {
+      value = value && value.split(',')
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:977]: we have modern javascript syntax now! enjoy it!
```suggestion
      value = value?.split(',')
```

File: packages/dd-trace/test/config.spec.js
Comment (simon-id) [inline packages/dd-trace/test/config.spec.js:1]: i believe there are *at least* two more tests to update here, the one that test for env var, and the one for programmatic config

**Review by simon-id — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/appsec/remote_config/capabilities.js (hunk 3 above)
Comment (ida613) [reply packages/dd-trace/src/appsec/remote_config/capabilities.js:19]: good catch

File: packages/dd-trace/src/config.js [hunk 6]
```diff
@@ -959,19 +955,38 @@ class Config {
...
     }
   }
 
+  _setSamplingRule (obj, name, value) {
+    if (value == null) {
+      return this._setValue(obj, name, null)
+    }
+
+    if (typeof value === 'string') {
+      value = value?.split(',')
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:977]: also btw i just realized (should have said that in the previous comment), if `typeof value === 'string'`, then how could value.split be undefined ? we know it's a string sooo it should have split right ?

Comment (simon-id) [reply packages/dd-trace/src/config.js:977]: for example, no check is made here: 
image

Comment (ida613) [reply packages/dd-trace/src/config.js:977]: true that

File: packages/dd-trace/test/config.spec.js [hunk 7]
```diff
@@ -662,12 +679,7 @@ describe('Config', () => {
       clientIpHeader: 'x-true-client-ip',
       sampleRate: 0.5,
       rateLimit: 1000,
-      samplingRules: [
-        { service: 'usersvc', name: 'healthcheck', sampleRate: 0.0 },
-        { service: 'usersvc', sampleRate: 0.5 },
-        { service: 'authsvc', sampleRate: 1.0 },
-        { sampleRate: 0.1 }
-      ],
+      samplingRules: samplingRules,
```
Comment (simon-id) [inline packages/dd-trace/test/config.spec.js:682]:
```suggestion
      samplingRules,
```

**Review by simon-id — COMMENTED** [review]
LGTM but wait for APM review too

**Review by simon-id — DISMISSED** [review]

**Review by bengl — APPROVED** [review]

**Pull request merged**